### DISCUSSION TITLE
Any way to wait for server closed?

### DISCUSSION THREAD

**Discussion by howesteve** [nanoexpress/nanoexpress#236] [Q&A] [answered]
Hello, this is more of a doubt. 

So when I call this:
```js
// ...
app.listen(3000))
```
.. a promise is returned, and even if I wait for it:
```js
await app.listen(3000))
```
the next line is executed immediately.

My question is, how can I wait until the server is stopped to finish closing other resources? 
```js
try {
  //...
  await app.listen(3000))
  // ==> should be waiting here somehow until server stops listening; e.g.
  await app.closed() 
} finally (e) {
  // cleaning up resources
  databasePool.close()
  app.close()
}
```
Thanks

**Comment by dalisoft** [comment, marked as answer]
Hi @howesteve 
Thank you for asking these questions, but it's not bug.

- There no `app.closed()` method
- `app.listen()` async/promise works properly, tested
- `app.close()` method code is [here](https://github.com/nanoexpress/nanoexpress/blob/master/src/App.js#L196)

**Reply by howesteve** [reply]
Thanks. I never thought this is a bug. However, after listen() the execution should pause until the server exits, is that right? 
But it's continuing right away. Let's see an example:
``` js

const app = require("nanoexpress")();
(async () => {
  await app.listen(3000);
  console.log(`I shouldn't print yet`); // it prints "I shouldn't print yet" immediately
})();
```
I was expecting the execution flow to pause at app.listen().

**Reply by dalisoft** [reply]
It waits until server starts. That at core level
